code/go/0chain.net/miner/protocol_block.go

```diff
@@ -624,6 +624,7 @@ func (mc *Chain) FinalizeBlock(ctx context.Context, b *block.Block) error {
 	for idx, txn := range b.Txns {
 		modifiedTxns[idx] = txn
 	}
+	b.FinalisedAt = common.Now()
```

Comment:
Different sharders will have different value of `common.Now()`, which will result in different `block.FinalizedAt` value. Basically every sharder has its own finalized time. Why do we need to save it to db, as if not necessary, I would suggest to not save inconsistent data.

Reply:
hroughout the lifetime of a block, we emit block events to events db at 2 places:
* when the block is notarized
* when the block is finalised
to differentiate between both the type of block events we would like to have a field that identify the finalised blocks. We could also use a boolean field. What do you suggest? shall I make it a boolean field?